meta_aggregation_api/services/limit_orders.py

```diff
@@ -60,28 +70,28 @@ async def get_by_wallet_address(
         )
         return res
 
-    @staticmethod
     async def get_by_hash(
+        self,
         chain_id: int,
         order_hash: str,
         provider: Optional[str],
     ):
         provider_class = all_providers.get(provider)
         if not provider_class:
             raise ProviderNotFound(provider)
-        provider_instance = provider_class()
+        provider_instance = provider_class(self.session, self.config, self.apm_client)
         logger.info(
             f'Getting limit order by hash: {order_hash}',
             extra={'provider': provider.__class__.__name__, 'order_hash': order_hash},
         )
-        res = await provider_instance.get_order_by_hash(
+        res = await provider_instance.get_order_by_hash(  # TODO: add the method to the base class?
```

Comment:
@shestakoven 

This method is not a member of the BaseProvider ABC. We should either add it to the BaseProvider or perform a type check on the provider instance.

Reply:
that's the method used only for limit orders type of provider... it's not needed for the rest of the providers. let's add to Base

Reply:
Why not just remove the `provider` parameter from the method and explicitly use LimitOrgerService class ?